File: samples/python/semantic-kernel-multiturn/src/plugins/date_time_plugin.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from datetime import datetime

from semantic_kernel.functions import kernel_function


class DateTimePlugin:
    @kernel_function(name="date", description="Get the current date.")
    def date(self) -> str:
        return datetime.now().strftime("%A, %B %d, %Y")

    @kernel_function(name="today", description="Get the current date.")
    def today(self) -> str:
        return self.date()

    @kernel_function(
        name="now",
        description="Get the current date and time in the local time zone.",
    )
    def now(self) -> str:
        return datetime.now().strftime("%A, %B %d, %Y %I:%M %p")

File: samples/dotnet/semantic-kernel-multiturn/Plugins/DateTimePlugin.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Microsoft.SemanticKernel;
using System;
using System.ComponentModel;

namespace SemanticKernelMultiturn.Plugins;

/// <summary>
/// Semantic Kernel plugins for date and time.
/// </summary>
public class DateTimePlugin
{
    [KernelFunction, Description("Get the current date.")]
    public string Date() => DateTimeOffset.Now.ToString("D");

    [KernelFunction, Description("Get the current date.")]
    public string Today() => this.Date();

    [KernelFunction, Description("Get the current date and time in the local time zone.")]
    public string Now() => DateTimeOffset.Now.ToString("f");
}
